File: solutions/go/difference-of-squares/2/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	sum := (1 + n) * n / 2
	return sum * sum
}

func SumOfSquares(n int) (result int) {
	return n * (1 + n) * (2*n + 1) / 6
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: solutions/ocaml/difference-of-squares/1/difference_of_squares.ml
let square_of_sum (n : int) : int = let sum = (1 + n) * n / 2 in sum * sum

let sum_of_squares (n : int) : int = n * (1 + n) * (2 * n + 1) / 6

let difference_of_squares (n : int) : int = square_of_sum(n) - sum_of_squares(n)
